src/oscar/apps/partner/strategy.py

```diff
@@ -211,7 +211,10 @@ def availability_policy(self, product, stockrecord):
         if not stockrecord:
             return Unavailable()
         if not product.get_product_class().track_stock:
-            return Available()
+            if stockrecord.price is None:
```

Comment:
I don't think this is the right solution. Even when the price is empty on a stockrecord, the product could still have a price through the strategy.

The purchaseinfo "object" has a price, availability and a stockrecord
https://github.com/django-oscar/django-oscar/blob/master/src/oscar/apps/partner/strategy.py#L14

The price is any of those (assuming a project has no custom ones)
https://github.com/django-oscar/django-oscar/blob/master/src/oscar/apps/partner/prices.py

So whenever a product has no price (even through the strategy) it will have the Unavailable price object
https://github.com/django-oscar/django-oscar/blob/master/src/oscar/apps/partner/prices.py#L40

So I see two options

1. Change the template, to make sure it checks whether a price exists
https://github.com/django-oscar/django-oscar/blob/master/src/oscar/templates/oscar/catalogue/partials/add_to_basket_form.html#L8
... and session.price.exists

2. Pass the price as argument in this method
https://github.com/django-oscar/django-oscar/blob/master/src/oscar/apps/partner/availability.py#L36

Which you would have to do here: https://github.com/django-oscar/django-oscar/blob/master/src/oscar/apps/partner/strategy.py#L126

And check session.price.exists here instead.

I think solution 2 would be cleaner, as it doesn't require a template change but it does require a change in the strategy. 

@specialunderwear what do you think?

Reply:
I think that the strategy should not be changed to pass the price because if needed we can just acces the price via ``self.pricing_policy`` so I think that the template must be changed to have better defaults. @joeyjurjens